### ISSUE TITLE
test_pg_regress is unstable on postgres 17: `invalid page in block 0 of relation`

### ISSUE THREAD

**Issue by jcsp** [neondatabase/neon#9240]
Example failure:
https://neon-github-public-dev.s3.amazonaws.com/reports/main/11141079169/index.html#testresult/b898a7720eeac6fa/retries

```
diff -U3 /__w/neon/neon/vendor/postgres-v17/src/test/regress/expected/rowsecurity.out /tmp/test_output/test_pg_regress[debug-pg17-None]-1/regress/results/rowsecurity.out
--- /__w/neon/neon/vendor/postgres-v17/src/test/regress/expected/rowsecurity.out	2024-10-02 09:24:30.001643081 +0000
+++ /tmp/test_output/test_pg_regress[debug-pg17-None]-1/regress/results/rowsecurity.out	2024-10-02 09:37:50.920969596 +0000
@@ -4172,6 +4172,7 @@
 SET SESSION AUTHORIZATION regress_rls_alice;
 SET row_security = on;
 CREATE TABLE r1 (a int);
+WARNING:  invalid page in block 0 of relation base/16384/1249_vm; zeroing out page
 INSERT INTO r1 VALUES (10), (20);
 CREATE POLICY p1 ON r1 USING (false);
 ALTER TABLE r1 ENABLE ROW LEVEL SECURITY;
```

**Comment by kelvich** [comment]
@knizhnik is actively investigating it. Hard to repro, but possible. More details: https://neondb.slack.com/archives/C04DGM6SMTM/p1727872045252899

**Comment by knizhnik** [comment]
I know the source of this zero VM pages:
```
                // Tail of last remaining vm page has to be zeroed.
                // We are not precise here and instead of digging in VM bitmap format just clear the whole page.
                modification.put_rel_page_image_zero(rel, vm_page_no)?;
```
So when performing TRUNCATE, we just zero VM page because it was to hard to do it accurately.
In principle we can just leave page header. But I do not think that it is critical.
I suggest to merge https://github.com/neondatabase/neon/pull/9287 and close this issue.

**Comment by knizhnik** [comment]
There is one more PR related with this problem: https://github.com/neondatabase/neon/pull/9342
which I want to be merged, although #9287 is enough to eliminate test failures.

**Comment by ololobus** [comment]
All related compute PRs have been merged, and it doesn't show up in the flaky top anymore, so closing
![Image](https://github.com/user-attachments/assets/e804879c-00b6-456e-8867-62348ef58dc9)
